components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -2387,7 +2440,24 @@ public IdentityProvider updateIdPByResourceId(String resourceId, IdentityProvide
                 return null;
             }
         }
-        return dao.getUpdatedIdPByResourceId(resourceId, tenantId, tenantDomain);
+
+        IdentityProvider identityProvider = dao.getUpdatedIdPByResourceId(resourceId, tenantId, tenantDomain);
+
+        // Validate whether the IdP satisfies the minimum requirements to be a JWT Issuer.
+        // Else remove existing JWT issuer entry.
+        if (isJwtIssuer(identityProvider)) {
+            // If the current IdP is already a JWT issuer, don't need to add again.
+            if (!isJwtIssuer(currentIdentityProvider)) {
+                dao.addAsAJWTIssuer(identityProvider, tenantId);
+            }
+        } else {
+            // If the current IdP is not a JWT issuer, don't need to remove again.
+            if (isJwtIssuer(currentIdentityProvider)) {
+                dao.removeAsAJWTIssuer(identityProvider, tenantId);
+            }
+        }
```

Comment:
shouldn't we just manipulate the identityProvider object here and let the DAO layer do the update accordingly?